lib/ecto/query.ex

```diff
@@ -989,6 +990,44 @@ defmodule Ecto.Query do
     for {k, v} <- binds, do: {{k, [], nil}, v}
   end
 
+  defp quoted_with_binds([{:limit, expr}, {:with_ties, wt} | t], quoted, binds, false) do
+    quoted =
+      quote do
+        Ecto.Query.limit(unquote(quoted), unquote(binds), unquote(expr), unquote(with_ties: wt))
+      end
+
+    {quoted, t}
+  end
+
+  defp quoted_with_binds([{:limit, expr}, {:with_ties, wt} | t], quoted, binds, true) do
+    quoted =
+      quote do
+        query = unquote(quoted)
+        Ecto.Query.limit(query, unquote(binds), unquote(expr), unquote(with_ties: wt))
+      end
+
+    {quoted, t}
+  end
+
+  defp quoted_with_binds([{type, expr} | t], quoted, binds, false) do
+    quoted =
+      quote do
+        Ecto.Query.unquote(type)(unquote(quoted), unquote(binds), unquote(expr))
+      end
+    
+    {quoted, t}
+  end
+
+  defp quoted_with_binds([{type, expr} | t], quoted, binds, true) do
+    quoted =
+      quote do
+        query = unquote(quoted)
+        Ecto.Query.unquote(type)(query, unquote(binds), unquote(expr))
+      end
+    
+    {quoted, t}
+  end
```

Comment:
Can we implement this like we implement `on` for consistency? Basically, as soon as you see a `:limit`, you call `{with_ties, t} = collect_limit(t)`.